app/src/main/java/fr/free/nrw/commons/nearby/DirectUpload.java

```diff
@@ -0,0 +1,95 @@
+package fr.free.nrw.commons.nearby;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.Build;
+import android.support.v4.app.Fragment;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AlertDialog;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import fr.free.nrw.commons.R;
+import fr.free.nrw.commons.contributions.ContributionController;
+
+import static android.Manifest.permission.READ_EXTERNAL_STORAGE;
+import static android.Manifest.permission.WRITE_EXTERNAL_STORAGE;
+import static android.content.pm.PackageManager.PERMISSION_GRANTED;
+
+class DirectUpload {
+
+    private String title;
+    private String desc;
+    private ContributionController controller;
+    private Fragment fragment;
+
+    @Inject @Named("prefs") SharedPreferences prefs;
+    @Inject @Named("default_preferences") SharedPreferences defaultPrefs;
+
+    DirectUpload(String title, String desc, Fragment fragment, ContributionController controller) {
+        this.title = title;
+        this.desc = desc;
+        this.fragment = fragment;
+        this.controller = controller;
+    }
+
+    void storeSharedPrefs() {
```

Comment:
This is a creative method of getting around the problem of not having a *shared model* - I applaud the creativity but it sets alarm bells ringing for me on an *architectural* level.  Instead of co-opting `SharedPreferences` as your data model, you might consider creating an "upload model" that you initialize as a singleton (using Dagger, in one of the modules) and inject into the various classes that need it.  I worry that the `SharedPreferences` approach would leave data hanging around and isn't as easily documented.